docs/tutorial.md

```diff
@@ -746,54 +777,77 @@ you will receive (value will vary):
 uptime_in_seconds:54782
 ```
 
-We assume that when the uptime is less than 60 seconds, the Redis service has recently started. Let's create a new function called `populateEvents`. It will call the `redis-cli info | grep uptime_in_seconds:` command and then create a notification event if the uptime value is smaller than a defined limit. To do so, we will use the method `AddNotificationEvent`, which adds a new event with the default `notifications` category for an integration object. It accepts the `string` argument, which is a summary message, i.e. `"Redis Server recently started"`.
+We assume that when the uptime is less than 60 seconds, the Redis service has recently started. We will call the `redis-cli info | grep uptime_in_seconds:` command and then create a notification event if the uptime value is smaller than a defined limit. To do so, we will use the type of event `event.NewNotification`, which is a event with the default `notifications` category for an integration object. It accepts the `string` argument, which is a summary message, i.e. `"Redis Server recently started"`.
 
 ```go
-func populateEvents(integration *sdk.Integration) error {
-	cmd := exec.Command("/bin/sh", "-c", "redis-cli info | grep uptime_in_seconds:")
-	output, err := cmd.CombinedOutput()
-	if err != nil {
-		return err
-	}
-	splittedLine := strings.Split(string(output), ":")
-	if len(splittedLine) != 2 {
-		return fmt.Errorf("Cannot split the output line")
-	}
-	uptime, err := strconv.ParseFloat(strings.TrimSpace(splittedLine[1]), 64)
-	if err != nil {
-		return err
-	}
-	if uptime < 60 {
-		err = integration.AddNotificationEvent("Redis Server recently started")
-	}
-
-	return err
+func main() {
+    // ...
+    // code for creating the integration and entity omitted
+    // code for populating Inventory and Metrics omitted
+    // ...
+    if args.All() || args.Events {
+        uptime, err := queryRedisInfo("uptime_in_seconds:")
+        panicOnErr(err)
+        if uptime < 60 {
+            err = entity.AddEvent(event.NewNotification("Redis Server recently started"))
+        }
+        panicOnErr(err)
+    }
 }
 ```
 
-Then, update `main()` function including `populateEvents` function.
+Then, update `main()` function including the snippet above that add events.
 
 ```go
 func main() {
-	integration, err := sdk.NewIntegration(integrationName, integrationVersion, &args)
-	fatalIfErr(err)
+	// Create Integration
+	i, err := integration.New(integrationName, integrationVersion, integration.Args(&args))
+	panicOnErr(err)
+
+	// Create Entity, entities name must be unique
+	entity := i.LocalEntity()
+	panicOnErr(err)
 
+	// Add Event
+	if args.All() || args.Events {
+		uptime, err := queryRedisInfo("uptime_in_seconds:")
+		panicOnErr(err)
+		if uptime < 60 {
+			err = entity.AddEvent(event.NewNotification("Redis Server recently started"))
+		}
+		panicOnErr(err)
+		if uptime < 60 {
+			err = entity.AddEvent(event.New("Redis Server recently started", "redis-server"))
```

Comment:
This `Event` doesn't belong in this example, you're adding it in the next one to show how to specify the category. If you look at the resulting json below, the event is not there.

Reply:
nice catch!